common/common-api/src/main/java/org/ow2/proactive/utils/Lambda.java

```diff
@@ -103,6 +104,27 @@ public static Runnable silent(RunnableThatThrows functionThatThrows) {
         };
     }
 
+    public static <T> void forEachWithIndex(Iterable<T> iterable, IteratorWithIndex<T> iteratorWithIndex) {
+
+        Iterator<T> iterator = iterable.iterator();
+
+        int index = 0;
+
+        while (iterator.hasNext()) {
+            T nextT = iterator.next();
+            iteratorWithIndex.forEachWithIndex(nextT, index);
+            index++;
+        }
+    }
+
+    private static <T, U, V> boolean allHaveNext(Iterator<T> iteratorT, Iterator<U> iteratorU, Iterator<V> iteratorV) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "allHaveNext" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)